tasks/runtime.yml

```diff
@@ -20,27 +20,55 @@
     mode: 0755
     state: directory
 
-- name: Generate a Munge key for the platform
+# Munge key handling is slightly complex as trying to cover cases with the following in the play:
+# a. entire cluster (standard ansible deployment)
+# b. only control node (part of image-based deploy)
+# c. only single compute node (for image building)
+# Therefore cannot depend on being able to generate a key on the control node and slurp + redistribute.
+# A local key can be generated from the path {{ openhpc_munge_key }}, 
+
+- name: Write Munge key from local disk
+  copy:
+    src: "{{ openhpc_munge_key }}"
+    dest: "/etc/munge/munge.key"
+    owner: munge
+    group: munge
+    mode: 0400
+  when: openhpc_munge_key is defined
+  notify:
+    - Restart Munge service
+
+- name: Generate a Munge key
+  # NB this is usually a no-op as the package install actually generates a (node-unique) one, so won't usually trigger handler
   command: "dd if=/dev/urandom of=/etc/munge/munge.key bs=1 count=1024"
   args:
     creates: "/etc/munge/munge.key"
-  when: inventory_hostname == openhpc_slurm_control_host
   notify:
     - Restart Munge service
 
-- name: Retrieve Munge key from Slurm control host
+- name: Retrieve Munge key
```

Comment:
Why do we need to retrieve the key from all hosts? Wondering if we should skip the slurping logic if `openhpc_munge_key is defined`

Reply:
Because this makes the modes build-entire-cluster, only-build-headnode, and only-build-compute-image (w/o a headnode) all the same